client/src/ingredients/attributes.js

```diff
@@ -16,31 +16,41 @@ function RenderValues({ values }) {
 }
 
 function RenderAttributes({ attributes }) {
-  return (
-    <dl>
-      {attributes.map(attribute => {
-        return (
-          <React.Fragment key={attribute.name}>
-            <dt>
-              <p><code>{attribute.name}</code>: <i>{attribute.type}</i></p>
-            </dt>
-            <dd>
-              {/* XXX a div tag in the middle of a dd tag!
-              See https://github.com/peterbe/mdn2/issues/5
-              */}
-              <div
-                dangerouslySetInnerHTML={{ __html: attribute.description }}
-              />
-
-              {attribute.values && attribute.values.length && (
-                <RenderValues values={attribute.values} />
-              )}
-            </dd>
-          </React.Fragment>
-        );
-      })}
-    </dl>
-  );
+  if (attributes.length === 0) {
+    return (
+      <p>This element only supports the <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes">
+        global attributes</a>.</p>
```

Comment:
This is oddly specific to "HTML" but surely the `RenderAttributes` component isn't HTML specific and will be reused for other types of content such as JavaScript and CSS. 

Is there a way to push this back to stumptown? It would be really cool if mdn2 is without any English content. 

After all, we have this: https://github.com/mdn/stumptown-experiment/blob/9b706592a12a1c79f6a6e1ced9e9e7d134f3d0c3/content/html/elements/abbr/meta.yaml#L17 but https://github.com/mdn/stumptown-experiment/tree/master/content/html/global_attributes doesn't yet exist. 

I'd rather we do something like:

```jsx
export function Attributes({ document }) {
  if (!(document.attributes && document.attributes.length)) {
    // TODO We need to support displaying a link to "global attributes" 
    // when a document doesn't have any. 
    // https://github.com/peterbe/mdn2/issues/XXX
    return null;
  }
  return (
    <>
      <h2>Attributes</h2>
      <RenderAttributes attributes={document.attributes} />
    </>
  );
}
```

Reply:
That'd unbreak the builder at the moment and it would be more correct.

Reply:
> This is oddly specific to "HTML" but surely the RenderAttributes component isn't HTML specific and will be reused for other types of content such as JavaScript and CSS.

This is specific to HTML. CSS and JS don't have attributes.

Reply:
Interesting. Then we should write that as a code comment. 

Or, do you want to attempt to resolve the [`.attributes.global`](https://github.com/mdn/stumptown-experiment/blob/9b706592a12a1c79f6a6e1ced9e9e7d134f3d0c3/content/html/elements/abbr/meta.yaml#L17) thing now?

Reply:
> After all, we have this: https://github.com/mdn/stumptown-experiment/blob/9b706592a12a1c79f6a6e1ced9e9e7d134f3d0c3/content/html/elements/abbr/meta.yaml#L17 but https://github.com/mdn/stumptown-experiment/tree/master/content/html/global_attributes doesn't yet exist.

Yes, I think eventually (once https://github.com/mdn/stumptown-experiment/tree/master/content/html/global_attributes does exist) then the global attributes will be present in the built JSON. This would be needed, for example, for an editor to be able to show all the attributes you could set for an element.

*But* that doesn't mean MDN will want to display them all on every page - that's a rendering decision. It might (as it currently does) want to just display a link to a separate page that lists them all. This tends to make me think that this text does live in mdn2 after all and not in stumptown.

> It would be really cool if mdn2 is without any English content.

I don't think this is practical. For example, look at BCD: https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow#Browser_compatibility. The table contain many strings that aren't in the compat data, such as: `"Experimental. Expect behavior to change in the future"`. I think these strings should not live in stumptown. It's the consumer's decision how to represent `experimental: true` in the way that's best suited for their application.

Reply:
> Interesting. Then we should write that as a code comment.

Sorry, what is "that"? The fact that attributes are HTML-specific or the fact that we will later have global attributes in the JSON?

Perhaps we should rename "RenderAttributes" to "RenderHTMLAttributes"?

> Or, do you want to attempt to resolve the [`.attributes.global`](https://github.com/mdn/stumptown-experiment/blob/9b706592a12a1c79f6a6e1ced9e9e7d134f3d0c3/content/html/elements/abbr/meta.yaml#L17) thing now?

I don't want to write the global attributes now. Write

Reply:
>>    It would be really cool if mdn2 is without any English content.

> I don't think this is practical. 

Fair enough. I did say "would be cool". Either way, let's attempt to keep it at a minimum unless it's a very functional string. I.e. strings of text that aren't web dev related.

Reply:
> Sorry, what is "that"? The fact that attributes are HTML-specific or the fact that we will later have global attributes in the JSON?

I meant, the factoid that on HTML elements have "Attributes" to render. It actually makes sense when I think about it because CSS doesn't really have that concept and in JavaScript objects have "methods", not attributes. 

Are we not going to add a page called things like `html/global_attributes/prose.md` etc ?

Reply:
> Are we not going to add a page called things like html/global_attributes/prose.md etc ?

Oh yes, for sure. Or something like that anyway. But not right now because I'm more focused on the rendering side.

Reply:
> I meant, the factoid that on HTML elements have "Attributes" to render. It actually makes sense when I think about it because CSS doesn't really have that concept and in JavaScript objects have "methods", not attributes.

Well we could I suppose. But they're just part of the technology. I mean would we also add comments that JS has methods to render, or CSS has properties? I would not have thought of doing that.